packages/contracts/contracts/foundry/SphinxUtils.sol

```diff
@@ -1295,4 +1254,42 @@ contract SphinxUtils is SphinxConstants {
 
         return _deploymentInfo;
     }
+
+    function concatJsonPath(string memory a, string memory b) public pure returns (string memory) {
+        return string(abi.encodePacked(a, ".", b));
+    }
+
+    function fetchProjectFromLock(address _script) public returns (SphinxLockProject memory) {
+        UserSphinxConfig memory _config = fetchAndValidateConfig(_script);
+        string memory root = vm.projectRoot();
+        string memory path = string(abi.encodePacked(root, "/sphinx.lock"));
+        string memory json = vm.readFile(path);
+
+        string memory basePath = concatJsonPath(".projects", _config.projectName);
+        bool exists = vm.keyExists(json, basePath);
+        if (!exists) {
+            revert(
+                string(
+                    abi.encodePacked(
+                        "Project with the name ",
+                        bytes(_config.projectName),
+                        " was not found in the `sphinx.lock` file. You need to register this project in the sphinx UI and then run `npx sphinx sync` to generate the latest `sphinx.lock` file. We recommend committing this file to version control."
```

Comment:
I don't think it's irrelevant. I included this because I assume the user will inevitably register a project and forget to run `sphinx sync` and commit the changes to version control.

You correctly point that out in this [comment](https://github.com/sphinx-labs/sphinx/pull/1667#discussion_r1577113767).

In this case, the following scenarios can occur:
1. The user proposes using the newly registered project. This works just fine since we fetch the `sphinx.lock` file from the backend during the proposal.
2. The user runs the deploy CLI command using the newly registered project or runs a test that utilizes the `sphinx` modifier. This fails with an error indicating that they should run `sphinx sync` and reminding them that this file should be committed to version control.